src/clj/kixi/hecuba/data/measurements/calculations.clj

```diff
@@ -242,3 +244,193 @@
   (let [{:keys [start-date end-date]} range]
     (doseq [timestamp (time/seq-dates start-date end-date (t/days 1))]
       (calculate-batch store sensor (t/minus timestamp (t/weeks 4)) (t/plus timestamp (t/days 1)) calculation-type))))
+
+(defn read-tariffs
+  "Reads profiles from the database and extracts tariffs."
+  [store entity_id]
+  (db/with-session [session (:hecuba-session store)]
+    (let [profiles (profiles/get-profiles entity_id session)]
+      (mapcat #(hash-map (:timestamp %) (-> % :profile_data :tariff)) profiles))))
```

Comment:
This `(mapcat #(hash-map ...)` looks odd, this is presumably to return a seq of pairs?  `([ts1 tariff1][ts2 tarrif2]` ?

Why not 
``` clojure
(map #(vector (:timestamp %) (-> % :profile_data :tariff)) profiles)
```